src/header.wppl

```diff
@@ -531,7 +531,7 @@ var Vector = function(arr) {
 var Matrix = function(arr) {
   var n = arr.length;
   var m = arr[0].length;
-  var t = ad.tensor.fromScalars(_.flatten(arr));
+  var t = ad.tensor.fromScalars(_.flattenDeep(arr));
```

Comment:
I think the matrix is always 2D, and that something else has gone wrong if it's not. (Paul, correct me if I'm wrong.) In that case, flattening only one step could make errors easier to find since `fromScalars` will hopefully throw an error if it gets a list as input.